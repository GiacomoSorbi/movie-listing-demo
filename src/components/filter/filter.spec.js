import React from 'react'
import { mount } from 'enzyme'
import toJson from 'enzyme-to-json'

import Filter from './'

describe('Filter should', () => {
  it('exist', () => {
    expect(Filter).toBeDefined()
  })

  it('have a consistent structure', () => {
    const wrapper = mount(
      <Filter name={'pippi'} onChange={jest.fn()} value={'cat'} />,
    )
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
