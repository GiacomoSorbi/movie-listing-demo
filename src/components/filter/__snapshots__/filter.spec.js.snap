// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter should have a consistent structure 1`] = `
<Filter
  name="pippi"
  onChange={[MockFunction]}
  value="cat"
>
  <div
    className="movie-filter"
  >
    <input
      id="cat"
      onChange={[MockFunction]}
      type="checkbox"
    />
    <label
      htmlFor="cat"
    >
      pippi
    </label>
  </div>
</Filter>
`;
